### DISCUSSION TITLE
Got NET-0001 while using .net SDK

### DISCUSSION THREAD

**Discussion by LeeKevinXO** [deepgram/community#223] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

After several seconds of calling StartConnectionAsync, ConnectionClosed was called and reported NET-0001.
What does 'NET-0001' mean? I searched it through the whole repository, nothing found about NET-0001...

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

POST /v1/listen?punctuate=true&diarize=true&encoding=linear16&sample_rate=16000

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

f9cdf264-8011-41be-9a11-444a4787a6eb

### If possible, please attach an example audio file to reproduce the issue.

Real-time audio data collecting using NAudio. sample rate 16000, soundtrack 1, bitdepth 16.

**Comment by nikolawhallon** [comment, marked as answer]
A `NET-0001` error means Deepgram hasn't received any data from the client for some amount of time (and so Deepgram assumes the connection is maybe dead/problematic and kills it). Some info on it is here: https://developers.deepgram.com/docs/troubleshooting-websocket-data-and-net-errors-when-live-streaming-audio

Are you sending audio during that time? If there are periods of time when you want the connection to Deepgram to stay active, but you don't want to send audio through (e.g. microphone muting), the `KeepAlive` feature is helpful: https://developers.deepgram.com/reference/streaming#stream-keepalive
